sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -201,7 +201,16 @@ object JoinReorderDP extends PredicateHelper {
     nextLevel.toMap
   }
 
-  /** Build a new join node. */
+  /**
+   * Builds a new JoinPlan if the two given sides can be joined with some conditions.
+   * @param oneJoinPlan One side JoinPlan for building a new JoinPlan.
+   * @param otherJoinPlan The other side JoinPlan for building a new join node.
+   * @param conf SQLConf for statistics computation.
+   * @param conditions The overall set of join conditions.
+   * @param topOutput The output attributes of the final plan.
+   * @return Return a new JoinPlan if the two sides can be joined with some conditions. Otherwise,
```

Comment:
How about? 
> Returns a new JoinPlan if there exists at least one join condition involving references from both left and right. Otherwise, returns None.